azurecompute-arm/src/main/java/org/jclouds/azurecompute.arm/AzureComputeProviderMetadata.java

```diff
@@ -0,0 +1,98 @@
+
+import static org.jclouds.azurecompute.arm.config.AzureComputeProperties.OPERATION_POLL_INITIAL_PERIOD;
+import static org.jclouds.azurecompute.arm.config.AzureComputeProperties.OPERATION_POLL_MAX_PERIOD;
+import static org.jclouds.azurecompute.arm.config.AzureComputeProperties.OPERATION_TIMEOUT;
+import static org.jclouds.azurecompute.arm.config.AzureComputeProperties.TCP_RULE_FORMAT;
+import static org.jclouds.azurecompute.arm.config.AzureComputeProperties.TCP_RULE_REGEXP;
+import static org.jclouds.oauth.v2.config.CredentialType.CLIENT_CREDENTIALS_SECRET;
+import static org.jclouds.oauth.v2.config.OAuthProperties.RESOURCE;
+import static org.jclouds.oauth.v2.config.OAuthProperties.AUDIENCE;
+import static org.jclouds.oauth.v2.config.OAuthProperties.CREDENTIAL_TYPE;
+import static org.jclouds.oauth.v2.config.OAuthProperties.JWS_ALG;
+
+import java.net.URI;
+import java.util.Properties;
+import org.jclouds.providers.ProviderMetadata;
+import org.jclouds.providers.internal.BaseProviderMetadata;
+
+import com.google.auto.service.AutoService;
+
+@AutoService(ProviderMetadata.class)
+public class AzureComputeProviderMetadata extends BaseProviderMetadata {
+
+   public static Builder builder() {
+      return new Builder();
+   }
+
+   @Override
+   public Builder toBuilder() {
+      return builder().fromProviderMetadata(this);
+   }
+
+   public AzureComputeProviderMetadata() {
+      super(builder());
+   }
+
+   public static Properties defaultProperties() {
+      final Properties properties = AzureManagementApiMetadata.defaultProperties();
+      properties.setProperty(OPERATION_TIMEOUT, "60000");
+      properties.setProperty(OPERATION_POLL_INITIAL_PERIOD, "5");
+      properties.setProperty(OPERATION_POLL_MAX_PERIOD, "15");
+      properties.setProperty(TCP_RULE_FORMAT, "tcp_%s-%s");
+      properties.setProperty(TCP_RULE_REGEXP, "tcp_\\d{1,5}-\\d{1,5}");
+      properties.put("oauth.endpoint", "https://login.microsoftonline.com/oauth2/token");
+      properties.put(JWS_ALG, "RS256");
+      properties.put(AUDIENCE, "https://login.microsoftonline.com/oauth2/token");
+      properties.put(RESOURCE, "https://management.azure.com");
+      properties.put(CREDENTIAL_TYPE, CLIENT_CREDENTIALS_SECRET.toString());
+      return properties;
+   }
+
+   public AzureComputeProviderMetadata(final Builder builder) {
+      super(builder);
+   }
+
+   public static class Builder extends BaseProviderMetadata.Builder {
+
+      protected Builder() {
+         super();
+
+         id("azurecompute-arm")
+                 .name("Azure Resource Management ")
+                 .apiMetadata(new AzureManagementApiMetadata())
+                 .endpoint("https://management.azure.com/subscriptions/SUBSCRIPTION_ID")
+                 .homepage(URI.create("https://www.windowsazure.com/"))
+                 .console(URI.create("https://windows.azure.com/default.aspx"))
+                 .linkedServices("azureblob", "azurequeue", "azuretable")
```

Comment:
This should reference to *existing* providers that can be accessed using the same account.